modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp

```diff
@@ -1061,16 +1061,15 @@ class WaveInputDeviceInstance  : public InputDeviceInstance
         // BEATCONNECT MODIFICATION START (RELAY)
         if (wi.getIsRelay() && wi.isEnabled())
         {
-            // Reset the size to 0.
-            // realyBufferProcessor will set the correct size.
-            inputBuffer.setSize(inputBuffer.getNumChannels(), 0, false, true);
-
             // relayBufferProcessor is a shared resource.
             // This callback function will be set to nullptr when the Relay disconnects or the application is closed.
             std::unique_lock<std::mutex> lock(edit.engine.getDeviceManager().m_muRelayCallback);
 
             if (nullptr != edit.engine.getDeviceManager().relayBufferProcessor)
-                edit.engine.getDeviceManager().relayBufferProcessor(inputBuffer);
+            {
+                bool isReady = (recordingContext != nullptr && recordingContext->adjustSamples < numSamples);
+                edit.engine.getDeviceManager().relayBufferProcessor(inputBuffer, numSamples, isReady);
```

Comment:
Because of the offset in the previous comment, relayBufferProcessor should also take in a start bases on recordingContext->adjustSamples.

Reply:
That maybe so.
The recorded clip is almost perfect but there is still a tinny shift.
I wanted to manage that in a different ticket so we can get this out to the testers.